src/Microsoft.Health.Dicom.Core.UnitTests/Features/Store/Entries/DicomInstanceEntryReaderForSinglePartRequestTests.cs

```diff
@@ -80,5 +87,56 @@ public async Task GivenBodyPartWithValidContentType_WhenReading_ThenCorrectResul
                     Assert.Same(_stream, await item.GetStreamAsync(source.Token));
                 });
         }
+
+        [Fact]
+        public async Task GivenBodyPartWithValidContentTypeExceedLimit_ThrowsError()
+        {
+            DicomInstanceEntryReaderForSinglePartRequest dicomInstanceEntryReaderLowLimit = new DicomInstanceEntryReaderForSinglePartRequest(_seekableStreamConverter, CreateStoreConfiguration(1));
+
+            using var source = new CancellationTokenSource();
+
+            Stream stream = new MemoryStream();
+            stream.Write(Encoding.UTF8.GetBytes("someteststring"));
+            stream.Seek(0, SeekOrigin.Begin);
+
+            await Assert.ThrowsAsync<DicomFileLengthLimitExceededException>(
+                () => dicomInstanceEntryReaderLowLimit.ReadAsync(
+                DefaultContentType,
+                stream,
+                source.Token));
+
+        }
+
+        private IOptions<StoreConfiguration> CreateStoreConfiguration(long maxSize)
+        {
+            var configuration = Substitute.For<IOptions<StoreConfiguration>>();
+            configuration.Value.Returns(new StoreConfiguration
+            {
+                MaxAllowedDicomFileSize = maxSize,
+            });
+            return configuration;
+        }
+
+        private class TestSeekableStreamConverter : ISeekableStreamConverter
+        {
+            public async Task<Stream> ConvertAsync(Stream stream, CancellationToken cancellationToken = default)
+            {
+                Stream seekableStream = null;
+
+                if (!stream.CanSeek)
+                {
+                    seekableStream = new FileBufferingReadStream(stream, 1024 * 30000, null, Environment.GetEnvironmentVariable("ASPNETCORE_TEMP") ?? Path.GetTempPath());
```

Comment:
You don't really need to use the `FileBufferingReadStream` and in turn the new nuget package. You could also just use `MemoryStream` and copy everything into there for the test

Reply:
Still need the package for drain async, but did change to memory stream